src/main/java/io/pravega/connectors/flink/FlinkPravegaWriter.java

```diff
@@ -55,106 +73,464 @@
     // The pravega stream name to write events to.
     private final String streamName;
 
-    // The sink's mode of operation. This is used to provide different guarantees for the written events.
-    private PravegaWriterMode writerMode = PravegaWriterMode.ATLEAST_ONCE;
+    private final long txnTimeoutMillis;
+    private final long txnMaxTimeMillis;
+    private final long txnGracePeriodMillis;
 
-    // Following runtime parameters are populated when the flink sub-tasks are executed at the task managers.
+    // Writer interface to assist exactly-once and atleast-once functionality
+    private transient InternalWriter<T> writerImpl = null;
 
-    // The pravega writer client.
+    /** The pravega writerImpl client. Used only to create transactions. */
```

Comment:
Looks like you (or your IDE) was overzealous in renaming `writer` to `writerImpl`.